iceoryx2/src/config.rs

```diff
@@ -395,14 +395,24 @@ impl Config {
     /// is called after this function was called, no file will be loaded since the global default
     /// config was already populated.
     pub fn global_config() -> &'static Config {
-        if !ICEORYX2_CONFIG.is_initialized()
-            && Config::setup_global_config_from_file(unsafe {
+        if !ICEORYX2_CONFIG.is_initialized() {
+            match Config::setup_global_config_from_file(unsafe {
                 &FilePath::new_unchecked(DEFAULT_CONFIG_FILE)
```

Comment:
Can we put the default config in a standard location? This would be useful for the `iox2 config` CLI too.

Reply:
Agreed. See also https://github.com/eclipse-iceoryx/iceoryx2/issues/437#issuecomment-2395451559

I would create a separate PR for this though. This PR could be part of a v0.4.2 release

Reply:
See #442